atomic_reactor/plugins/post_export_operator_manifests.py

```diff
@@ -0,0 +1,179 @@
+from atomic_reactor.util import df_parser, is_scratch_build, get_platforms
+
+OPERATORS_LABEL = 'com.redhat.delivery.appregistry'
+MANIFESTS_DIR_NAME = 'manifests'
+OPERATORS_ARCHIVE_NAME = 'operator_manifests'
+OPERATORS_ARCHIVE_EXT = 'zip'
+IMAGE_MANIFEST_NAME = 'manifest.json'
+WHITEOUT_FILE_PREFIX = '.wh.'
+
+
+class ExportOperatorManifestsPlugin(PostBuildPlugin):
+    """
+    Export operator manifest files
+
+    Fetch and archive operator manifest files from image so they can be
+    uploaded to koji.
+    """
+
+    key = PLUGIN_EXPORT_OPERATOR_MANIFESTS_KEY
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow):
+        """
+        constructor
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        """
+        super(ExportOperatorManifestsPlugin, self).__init__(tasker, workflow)
+
+    def has_operator_manifest(self):
+        """
+        Check if the plugin is running in orchestrator.
+
+        :return: bool
+        """
+        dockerfile = df_parser(self.workflow.builder.df_path, workflow=self.workflow)
+        # TODO: set the label in util Labels
+        return dockerfile.labels.get(OPERATORS_LABEL, False)
+
+    def is_orchestrator(self):
+        """
+        Check if the plugin is running in orchestrator.
+
+        :return: bool
+        """
+        if get_platforms(self.workflow):
+            return True
+        return False
+
+    def should_run(self):
+        """
+        Check if the plugin should run or skip execution.
+
+        :return: bool, False if plugin should skip execution
+        """
+        # platform = os.uname()[4]
+        if self.is_orchestrator():
+            self.log.warning("%s plugin set to run on orchestrator. Skipping", self.key)
+            return False
+        # TODO: skip if not in selected platform
+        if is_scratch_build():
+            self.log.info("Scratch build. Skipping")
+            return False
+        if not self.has_operator_manifest():
+            self.log.info("Operator manifests label not set in Dockerfile. Skipping")
+            return False
+        return True
+
+    def run(self):
+        """
+        Run the plugin.
+        """
+        if not self.should_run():
+            return
+        saved_image = self.workflow.exported_image_sequence[-1].get('path')
+        self.log.debug("Extracting data from %s", saved_image)
+        temp_dir = tempfile.mkdtemp()
+        manifests_zipfile_path = self.extract_manifests(saved_image, temp_dir)
+        return manifests_zipfile_path
+
+    def get_archived_manifests(self, members):
+        for tarinfo in members:
+            if tarinfo.name.startswith(os.path.join(MANIFESTS_DIR_NAME, '')):
+                yield tarinfo
+
+    def extract_manifests(self, image_archive_path, workdir):
+        """
+        Extract operator manifest files from image layers
+
+        :param image_archive_path: str, path to docker image archive tar file
```

Comment:
Why use this instead of the equivalent of `docker export`? The current code ties us to the `docker save` format.

Reply:
would it be reasonable to use `get_archive` [1]?

[1]  https://docker-py.readthedocs.io/en/stable/api.html?highlight=get_image#docker.api.container.ContainerApiMixin.get_archive

Reply:
Perfect.